pkg/internal/pop/poptoken.go

```diff
@@ -81,23 +88,57 @@ func (swk *swKey) init(key *rsa.PrivateKey) {
 	swk.jwk = getJWK(eB64, nB64, swk.keyID)
 }
 
-// generateSwKey generates a new swkey and initializes it with required fields before returning it
-func generateSwKey(key *rsa.PrivateKey) (*swKey, error) {
-	swk := &swKey{}
+// generateSwKey generates a new SwKey and initializes it with required fields before returning it
+func generateSwKey(key *rsa.PrivateKey) (*SwKey, error) {
+	swk := &SwKey{}
 	swk.init(key)
 	return swk, nil
 }
 
-// GetSwPoPKey generates a new PoP key that rotates every 8 hours and returns it
-func GetSwPoPKey() (*swKey, error) {
+// GetUniqueSwPoPKey returns a new PoP key which is maintained and that rotates every 8 hours
+func GetUniqueSwPoPKey() (*SwKey, error) {
+	pwsKeyMutex.Lock()
+	defer pwsKeyMutex.Unlock()
+
+	if pswKey != nil {
+		return pswKey, nil
+	}
+
+	key, err := GetSwPoPKey()
+	if err != nil {
+		return nil, fmt.Errorf("error generating PoP key: %w", err)
+	}
+	pswKey = key
+
+	// rotates key every 8 hours
+	ticker := time.NewTicker(8 * time.Hour)
```

Comment:
In the context of kubelogin not sure if this make sense.
Kubelogin is a one shot binary invokation, you cannot maintain a go routine in the background for 8h.
For the kubelogin flow, we might want this to be unique all the time, if you need to change this behavior this will need to init/passed by the caller